extra_foam/gui/pyqtgraph/graphicsItems/GraphicsObject.py

```diff
@@ -129,7 +129,7 @@ def toLogScale(arr, policy=None):
         """Convert array result to logarithmic scale."""
         ret = np.nan_to_num(arr)
         ret[ret < 0] = 0
-        return np.log10(ret + 1)
+        return np.log10(ret, where=(ret > 0))
```

Comment:
I agree that the original solution does not work in some situations, as you have shown. However, there are also corner cases for the new one. It makes 1 and 0 the same. 0 will not be affected while 1 will become 0 after the log operation. Imaging you have a step function `f(x) = 1 if x > 0 else 0`, with the new solution the curve will become completely flat (all 0). Therefore, a better solution might be letting users choose among different methods.